src/api/dirmonitor.c

```diff
@@ -71,12 +71,12 @@ static int f_dirmonitor_new(lua_State* L) {
 
 static int f_dirmonitor_gc(lua_State* L) {
   struct dirmonitor* monitor = luaL_checkudata(L, 1, API_TYPE_DIRMONITOR);
-  SDL_LockMutex(monitor->mutex);
   monitor->length = -1;
-  deinit_dirmonitor(monitor->internal);
-  SDL_UnlockMutex(monitor->mutex);
   SDL_WaitThread(monitor->thread, NULL);
+  SDL_LockMutex(monitor->mutex);
+  deinit_dirmonitor(monitor->internal);
   free(monitor->internal);
+  SDL_UnlockMutex(monitor->mutex);
   SDL_DestroyMutex(monitor->mutex);
   return 0;
 }
```

Comment:
This would break the inotify backend.

inotify keeps a pipe open and relies on the write pipe sending an EOF when closing, causing the `poll` to continue and the read syscall to fail because the monitor fd is invalid.

The same approach could be implemented for kqueue, but I genuinely think that this is a terrible idea that should burn but reverting this would not break the kqueue backend, so I don't care.